src/main/webapp/js/finscholar/commonlistview/commonlistview.soy

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+{namespace finscholar.commonlistview.templates}
+
+{template .commonlistview}
+  {@param pageIndex : string}
+  <table class="mdl-data-table mdl-js-data-table 
+      mdl-data-table--selectable mdl-shadow--2dp">
+    <thead>
+      {if $pageIndex == '1'}
+        {call .scholarshiplistheader /}
+      {else}
+        {call .collegelistheader /}
+      {/if}
+    </thead>
+    <tbody id="table-body">
+    </tbody>
+  </table>
+  <div id="status">
+    {call .endoflist /}
+  </div>
+{/template}
+
+{template .scholarshiplistheader}
+  <tr id="title">
+    <th class="mdl-data-table__cell--non-numeric">Scholarship Name</th>
+    <th>Schools</th>
+    <th>Amount</th>
+  </tr>
+{/template}
+
+{template .collegelistheader}
+  <tr id="title">
+    <th class="mdl-data-table__cell--non-numeric">College Name</th>
+    <th>Acceptance Rate</th>
+    <th>Average ACT</th>
+  </tr>
+{/template}
+
+{template .scholarshiplistitems}
+  {@param scholarships: list<[scholarshipName: string, 
+                              schools: string, 
+                              amountPerYear: string, 
+                              id:string]>}
+  {for $item in $scholarships}
+    <tr class="scholarship" id={$item.id}
+        jsaction="scholarshiplistview.clickAction;
+        dblclick:scholarshiplistview.doubleClickAction">
+      <td class="mdl-data-table__cell--non-numeric">{$item.scholarshipName}</td>
+      <td>{$item.schools}</td>
+      <td>{$item.amountPerYear}</td>
+    </tr>
+  {/for}
+{/template}
+
+{template .collegelistitems}
+  {@param colleges: list<[name: string, 
+                          acceptanceRate: string, 
+                          averageACT: string, 
+                          id:string]>}
+  {for $item in $colleges}
+    <tr class="class" id={$item.id}
+        jsaction="collegelistview.clickAction;
+        dblclick:collegelistview.doubleClickAction">
+      <td class="mdl-data-table__cell--non-numeric">{$item.name}</td>
+      <td>{$item.acceptanceRate}</td>
+      <td>{$item.averageACT}</td>
+    </tr>
+  {/for}
+{/template}
```

Comment:
You can combine these since they're identical. Make a more generic structure for the data and treat the key and values as strings.